### PR TITLE
Ftr login

### PR DESCRIPTION
**Pull request by morellaj**
Added login page with login button
Added bootstrap dropdown with logout button to the navbar
Added username and user email states to app.jsx and context for those states
Updated testing to test for logging in and out

### CODE REVIEW

File: src/Components/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React, { useContext } from "react";
+import styled from "styled-components";
+import { UserContext } from "./UserContext";
+import { useGoogleLogin } from "react-google-login";
+
+export default function Login() {
+  const { setUsername, setUserEmail } = useContext(UserContext);
+  const { signIn } = useGoogleLogin({
+    clientId: "1084859424709-tk8745k1d0bnqfvlmsoa0j3uo5bkm9un.apps.googleusercontent.com",
```
Comment (alecheffner) [inline src/Components/Login.jsx:9]: Since this is used in multiple places lets put it in a single location

File: src/Components/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  align-items: center;
+  text-align: center;
+  flex-direction: column;
+`;
+
+const Welcome = styled.h2`
+  margin-bottom: 30px;
+`;
+
+const SignIn = styled.div`
```
Comment (alecheffner) [inline src/Components/Login.jsx:38]: add "cursor: pointer"

File: src/Components/Login.jsx (hunk 1 above)
Comment (morellaj) [reply src/Components/Login.jsx]: Agreed, for now moved it to "constants.js".  May want to move it to a configuration file or something like that in the future.

File: src/Components/Login.jsx (hunk 2 above)
Comment (morellaj) [reply src/Components/Login.jsx:39]: Done

**Pull request merged**